RecoParticleFlow/PFProducer/interface/TableDefinitions.h

```diff
@@ -0,0 +1,191 @@
+#ifndef RecoParticleFlow_PFProducer_TableDefinitions_h
+#define RecoParticleFlow_PFProducer_TableDefinitions_h
+#include <set>
+#include <vector>
+
+#include "FWCore/SOA/interface/Table.h"
+#include "FWCore/SOA/interface/TableView.h"
+#include "FWCore/SOA/interface/Column.h"
+
+#include "DataFormats/Provenance/interface/ElementID.h"
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/RefToBase.h"
+#include "DataFormats/ParticleFlowReco/interface/PFLayer.h"
+
+template <class C>
+edm::ElementID refToElementID(const edm::Ref<C>& ref) {
+  return edm::ElementID(ref.id(), ref.index());
+}
+
+template <class C>
+edm::ElementID refToElementID(const edm::RefToBase<C>& ref) {
+  return edm::ElementID(ref.id(), ref.key());
+}
```

Comment:
this is too generic to be in a  RecoParticleFlow/PFProducer; perhaps these also should be in the edm namespace